lib/src/rules/prefer_if_elements_to_map_literals.dart

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2019, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:linter/src/analyzer.dart';
+
+const _desc = r'Prefer if elements to map literals where possible.';
+
+const _details = r'''
+When building maps from iterables, it is preferable to use `if` and `for` 
+elements rather than conditionals and repetition.
```

Comment:
The assist doesn't use if elements, so I think we can drop that part. (That confusion was probably my fault. Looking at the assist I see I wrote "if" where I should have written "for".) Also, I don't know what "rather than conditionals and repetition" means in this context. How about something like:

When building maps from iterables, it is preferable to use `for` elements.

Reply:
Perfect.  Thanks!